homeassistant/components/sensor/bom_weather_current.py

```diff
@@ -0,0 +1,188 @@
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the BOM sensor."""
+    rest = BOMCurrentData(hass, config.get(CONF_ZONE_ID),
+                          config.get(CONF_WMO_ID))
+    sensors = []
+    for variable in config[CONF_MONITORED_CONDITIONS]:
+        sensors.append(BOMCurrentSensor(rest,
+                                        variable,
+                                        config.get(CONF_NAME)))
+
+    try:
+        rest.update()
+    except ValueError as err:
+        _LOGGER.error("Received error from BOM_Current: {}".format(err))
+        return False
+
+    add_devices(sensors)
+
+    return True
+
+
+class BOMCurrentSensor(Entity):
+    """Implementing the BOM current sensor."""
+
+    def __init__(self, rest, condition, stationname):
+        """Initialize the sensor."""
+        self.rest = rest
+        self._condition = condition
+        self.stationname = stationname
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self.stationname is None:
+            return "BOM {}".format(SENSOR_TYPES[self._condition][0])
+        else:
+            return "BOM {} {}".format(self.stationname,
+                                      SENSOR_TYPES[self._condition][0])
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.rest.data and self._condition in self.rest.data:
+            return self.rest.data[self._condition]
+        else:
+            return STATE_UNKNOWN
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attr = {}
+        attr['Sensor Id'] = self._condition
+        attr['Zone Id'] = self.rest.data['history_product']
+        attr['Station Id'] = self.rest.data['wmo']
+        attr['Station Name'] = self.rest.data['name']
+        attr['Last Update'] = datetime.datetime.strptime(str(
+                                self.rest.data['local_date_time_full']),
+                                '%Y%m%d%H%M%S')
+        return attr
+
+    @property
+    def unit_of_measurement(self):
+        """Return the units of measurement."""
+        return SENSOR_TYPES[self._condition][1]
+
+    def update(self):
+        """Update current conditions."""
+        self.rest.update()
+
+
+class BOMCurrentData(object):
+    """Get data from BOM."""
+
+    def __init__(self, hass, zone_id, wmo_id):
+        """Initialize the data object."""
+        self._hass = hass
+        self._zone_id = zone_id
+        self._wmo_id = wmo_id
+        self.data = None
+        self._lastupdate = LAST_UPDATE
+
+    def _build_url(self):
+        url = _RESOURCE.format(self._zone_id, self._zone_id, self._wmo_id)
+        _LOGGER.info("BOM url {}".format(url))
+        return url
```

Comment:
I thought so too. This process of building the url was borrowed from another component as I have tried to used what others are doing in components to keep the style in line with the project .